heka/files/lua/common/afd.lua

```diff
@@ -149,33 +148,29 @@ function inject_afd_metric(msg_type, msg_tag_name, msg_tag_value, metric_name,
         no_alerting = true
     end
 
+    local tag_fields = {'hostname', 'member'}
     local msg = {
-        Type = msg_type,
+        Type = 'afd_metric',
         Payload = payload,
         Fields = {
-            name=metric_name,
+            name='status',
             value=value,
             hostname=hostname,
-            interval=interval,
-            source=source,
-            tag_fields={msg_tag_name, 'source', 'hostname'},
-            no_alerting = no_alerting,
+            member=afd_name,
+            no_alerting=no_alerting,
         }
     }
-    msg.Fields[msg_tag_name] = msg_tag_value,
+
+    for dimension_key, dimension_value in pairs(dimensions) do
+        table.insert(tag_fields, dimension_key)
+        msg.Fields[dimension_key] = dimension_value
```

Comment:
this could overwrite existing (and potentially important, eg 'member') fields. Do we want to protect from this?